docs/creating-a-static-check.md

```diff
@@ -0,0 +1,165 @@
+  }
+}
+```
+
+Here are two more concrete ways to write static checks.
+
+## Creating a new Check using PHP CodeSniffer.
+
+A new Check class should be created for the static check and should implement the following attributes.
+
+- The class name should be suffixed with `_Check`.
+- The class should extend the `Abstract_PHP_CodeSniffer_Check` class.
+- The class should implement the `get_args()` method.
+
+## Creating a new Check using Abstract_File_Check.
+
+A new Check class should be created for the static check and should implement the following attributes.
+
+- The class name should be suffixed with `_Check`.
+- The class should extend the `Abstract_File_Check` class.
+- The class should implement the `check_files()` method.
+
+### PHPCS Arguments
+
+Rather than write all the logic required to run PHPCS, the `Abstract_PHP_CodeSniffer_Check` handles this logic so developers only need to supply the arguements to PHPCS in order to get their check working.
+
+This is done by implementing the `get_args()` method in the new Check class defined within the `Abstract_PHP_CodeSniffer_Check`.
+
+The `get_args()` method should return an associative array containing the PHPCS arguments including the file extension, code standard and specific sniff to run.
+
+Below is an example of a Static Check class that checks for i18n usage in the plugins codebase.
+
+```php
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Checker\Checks\Abstract_PHP_CodeSniffer_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+/**
+ * Check for running WordPress internationalization sniffs.
+ */
+class I18n_Usage_Check extends Abstract_PHP_CodeSniffer_Check {
+  
+  use Stable_Check;
+
+  /**
+   * Gets the categories for the check.
+   *
+   * Every check must have at least one category.
+   *
+   * @since n.e.x.t
+   *
+   * @return array The categories for the check.
+   */
+  public function get_categories() {
+    return array( Check_Categories::CATEGORY_GENERAL );
+  }
+
+  /**
+   * Returns an associative array of arguments to pass to PHPCS.
+   *
+   * @return array An associative array of PHPCS CLI arguments.
+   */
+  protected function get_args() {
+    return array(
+      'extensions' => 'php',
+      'standard'   => 'WordPress',
+      'sniffs'     => 'WordPress.WP.I18n',
+    );
+  }
+}
+```
+
+## Add the Check to the Plugin Checker
+
+In order to run the check as part of the Plugin Checker process it needs to be added to the Plugin Checkers list of available checks.
+
+This is done by using the `wp_plugin_check_checks` filter to register an instance of the check with its slug.
+
+- If you're contributing to the actual plugin checker, add the check to the list in the `Abstract_Check_Runner::register_checks()` method.
+- If you're implementing a check in code outside of the actual plugin checker, use the approach you've described here so far.
```

Comment:
This is confusing because you mention the filter in the beginning although it only applies to the second bullet point. Consider this instead:
```suggestion
- If you're contributing to the actual plugin checker, add the check to the list in the `Abstract_Check_Runner::register_checks()` method.
- If you're implementing a check in code outside of the actual plugin checker, use the `wp_plugin_check_checks` filter, as seen in the example below.
```

Reply:
Updated